packages/server/lib/browsers/cdp_automation.ts

```diff
@@ -174,18 +174,29 @@ const ffToStandardResourceTypeMap: { [ff: string]: ResourceType } = {
   'webmanifest': 'manifest',
 }
 
+interface CdpAutomationOptions {
+  onBrowserPreRequest?: (BrowserPreRequest) => void
+}
+
 export class CdpAutomation {
-  constructor (private sendDebuggerCommandFn: SendDebuggerCommand, onFn: OnFn, private automation: Automation) {
+  frameTree?: Protocol.Page.FrameTree
+
+  constructor (private sendDebuggerCommandFn: SendDebuggerCommand, onFn: OnFn, private automation: Automation, private options: CdpAutomationOptions = {}) {
     onFn('Network.requestWillBeSent', this.onNetworkRequestWillBeSent)
     onFn('Network.responseReceived', this.onResponseReceived)
-    sendDebuggerCommandFn('Network.enable', {
+  }
+
+  async enable () {
+    await this.sendDebuggerCommandFn('Network.enable', {
       maxTotalBufferSize: 0,
       maxResourceBufferSize: 0,
       maxPostDataSize: 0,
     })
+
+    await this.sendDebuggerCommandFn('Page.enable')
```

Comment:
should we skip enabling `Page` on FF and electron, since it won't be used?

actually, since `Page` is listened to outside this file, `Page.enable` doesn't seem like it belongs in `cdp_automation.ts` at all, since it's not generic to CDP, it's only used for Chromium. the changes to this file can probably be confined to `chrome.ts`

Reply:
[Page.captureScreenshot](https://github.com/cypress-io/cypress/blob/2bf3557a0b4d8650cd8880058288d510a514026e/packages/server/lib/browsers/cdp_automation.ts#L330) is used in `cdp_automation.ts`, so I thought it made sense to move it in here, but now I see that chrome is the only browser that uses that `onRequest` callback. I'll move it into `chrome.ts`.

Reply:
Gotcha. I think `Page.captureScreenshot` would still work, all `Page.enable` does is enable receiving `Page.*` events, I think it might even affect the `Page.screencastFrame` event.

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/19323/commits/bf0581282ca42388defdc73d08057f12b44e3eb0